### PR TITLE
fallback load model

### PR DESCRIPTION
**Pull request by jiqing-feng**
Hi @echarlaix . The `IPEXModel` has 2 loading methods:
1. `_from_transformers` accepts a model_path_or_id with an original transformers model and exports this model.
2. `_from_pretrained` only accepts an exported model; otherwise, it will raise ValueError.

I was thinking of falling back the  `_from_pretrained` to `_from_transformers` if the model has not been exported yet and logging a warning to alert users. WDYT?

Besides, check the model type before using `torch.jit.save` in `save_pretrained`.

### CODE REVIEW

**Comment by jiqing-feng** [comment]
Hi @echarlaix . As we discussed before, I have removed `_from_transformers`.

File: optimum/intel/ipex/modeling_base.py [hunk 1]
```diff
@@ -187,60 +197,42 @@ def _from_transformers(
...
+            if use_auth_token is not None:
+                warnings.warn(
+                    "The `use_auth_token` argument is deprecated and will be removed in v5 of Transformers. Please use `token` instead.",
+                    FutureWarning,
+                )
+                if token is not None:
+                    raise ValueError(
+                        "Both the arguments `use_auth_token` and `token` were specified, which is not supported. Please specify only `token`."
+                    )
+                token = use_auth_token
```
Comment (echarlaix) [inline optimum/intel/ipex/modeling_base.py:211]: I thik this should be outside (could be the case when loading a torchscript model on the hub)

File: optimum/intel/ipex/modeling_base.py [hunk 2]
```diff
@@ -160,9 +160,17 @@ def __init__(
             self._init_warmup()
 
     @classmethod
-    def _from_transformers(
+    def _from_transformers(cls, *args, **kwargs):
+        logger.warning(
+            "_from_transformers will be removed in the future, please remove the export arguments in the API."
+        )
```
Comment (echarlaix) [inline optimum/intel/ipex/modeling_base.py:166]: We should keep `export` to know when the model needs to be converted to torchscript no ? Especially if we want to have default compatibility across devices (xpu / cpu)

**Comment by jiqing-feng** [comment]
Hi @echarlaix . I have fixed all your comments, pls take a reviewn thx!

File: optimum/intel/ipex/modeling_base.py [hunk 3]
```diff
@@ -169,9 +176,17 @@ def __init__(
...
```
Comment (echarlaix) [inline optimum/intel/ipex/modeling_base.py:182]: I think we should remove this warning as  user still should load the model with :  `.from_pretrained(.., export=True)` 
```suggestion
```

**Review by echarlaix — APPROVED** [review]
LGTM, thanks @jiqing-feng

File: optimum/intel/ipex/modeling_base.py [hunk 4]
```diff
@@ -169,9 +176,14 @@ def __init__(
             self._init_warmup()
 
     @classmethod
-    def _from_transformers(
+    def _from_transformers(cls, *args, **kwargs):
+
```
Comment (echarlaix) [inline optimum/intel/ipex/modeling_base.py:180]:
```suggestion
```

**Comment by jiqing-feng** [comment]
Tests fixed!  @echarlaix

File: optimum/intel/ipex/modeling_base.py [hunk 5]
```diff
@@ -438,20 +424,22 @@ def __init__(
...
+        config.use_cache = use_cache
+        super().__init__(model, config, export, model_save_dir=model_save_dir, warmup=False)
         GenerationMixin.__init__(self)
 
         model_type = self.config.model_type.replace("_", "-")
         self.normalized_config = NormalizedConfigManager.get_normalized_config_class(model_type)(self.config)
         self.use_cache = "past_key_values" in self.input_names
 
-        if use_cache ^ self.use_cache:
+        if export and use_cache ^ self.use_cache:
```
Comment (echarlaix) [inline optimum/intel/ipex/modeling_base.py:442]:
```suggestion
        if isinstance(model, torch.jit.RecursiveScriptModule) and use_cache ^ self.use_cache:
```

File: optimum/intel/ipex/modeling_base.py [hunk 6]
```diff
@@ -438,20 +424,22 @@ def __init__(
...
+        export: bool = False,
         model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
         use_cache: bool = True,
         warmup: bool = True,
         **kwargs,
     ):
         # Perform the initial warmup at the end of __init__
-        super().__init__(model, config, model_save_dir=model_save_dir, warmup=False)
+        config.use_cache = use_cache
+        super().__init__(model, config, export, model_save_dir=model_save_dir, warmup=False)
```
Comment (echarlaix) [inline optimum/intel/ipex/modeling_base.py:435]:
```suggestion
        super().__init__(model, config, export=export, model_save_dir=model_save_dir, warmup=False)
```

File: optimum/intel/ipex/modeling_base.py [hunk 7]
```diff
@@ -144,11 +146,14 @@ def __init__(
...
-            config = model.config if config is None else config
-            use_cache = getattr(model.config, "use_cache", False)
-            model = ipex_jit_trace(model, self.export_feature, use_cache)
-            config.torchscript = True
+        if export:
+            if isinstance(model, torch.jit.RecursiveScriptModule):
+                logger.warning("The model has been exported already.")
+            else:
+                config = model.config if config is None else config
+                use_cache = kwargs.get("use_cache", None)
```
Comment (echarlaix) [inline optimum/intel/ipex/modeling_base.py:154]: actually should be set by default to True to fix compatibility with v1.17.0

https://github.com/huggingface/optimum-intel/blob/v1.17.2/optimum/intel/ipex/modeling_base.py#L159
```suggestion
                use_cache = kwargs.get("use_cache", True)
```

**Pull request merged**